core/trino-main/src/main/java/io/trino/sql/planner/plan/NonReadingTableExecuteNode.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.plan;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import io.trino.metadata.TableExecuteHandle;
+import io.trino.metadata.TableHandle;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.sql.planner.Symbol;
+
+import java.util.List;
+import java.util.Optional;
+
+public class NonReadingTableExecuteNode
+        extends PlanNode
+{
+    private final Symbol output;
+    private final TableExecuteHandle executeHandle;
+    private final Optional<TableHandle> sourceHandle;
+    private final SchemaTableName schemaTableName;
```

Comment:
is there a source handle?

how does this differ from `TableExecuteTarget` ?

Reply:
there is no source handle.
It doesn't differ, initially I had specific target class but Lukasz advised to keep this data inside this Node class

Reply:
> there is no source handle.

then shall we get rid of `Optional<TableHandle> sourceHandle` field?